### PR TITLE
Optimize fetching&parsing of activity logs for azure storage accounts

### PR DESCRIPTION
**Pull request by misg**
This PR optimizes the fetching and parsing of azure activity logs, used when checking for access key rotations regarding the storage accounts.
In particular, as suggested in #252, the fetching of activity logs is now done by using the `resourceUri` filter (e.g. by using storage accounts ids instead of pulling every existing activity logs).
As consequences, the monitor service is now useless and therefore is being removed in this PR.

### CODE REVIEW

File: ScoutSuite/providers/azure/rules/findings/storageaccount-access-keys-not-rotated.json [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {
     "dashboard_name": "Storage Accounts",
     "description": "Access Keys Not Rotated",
-    "rationale": "The access keys of your storage accounts should be rotated at least every 90 days. See CIS 3.3.",
+    "rationale": "The access keys of your storage accounts should be rotated at least every _ARG_0_ days. See CIS 3.3.",
```
Comment (j4v) [inline ScoutSuite/providers/azure/rules/findings/storageaccount-access-keys-not-rotated.json]: :clap: :clap: :clap:

**Comment by j4v** [comment]
*Brilliant*, this is going to run sooooo much faster!

File: ScoutSuite/providers/azure/facade/storageaccounts.py [hunk 2]
```diff
@@ -1,15 +1,56 @@
...
 
     async def get_blob_containers(self, resource_group_name, storage_account_name):
         return await run_concurrently(
             lambda: self._client.blob_containers.list(resource_group_name, storage_account_name).value
         )
+
+    async def get_and_set_activity_logs(self, storage_account):
+        client = MonitorManagementClient(self._credentials, self._subscription_id)
+
+        # time format used by Azure API:
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/facade/storageaccounts.py]:
```suggestion
        # Time format used by Azure API:
```

File: ScoutSuite/providers/azure/facade/storageaccounts.py [hunk 3]
```diff
@@ -1,15 +1,56 @@
...
+        time_format = "%Y-%m-%dT%H:%M:%S.%f"
+        # Azure API uses UTC time, we need to use the same to avoid bad requests:
+        utc_now = datetime.datetime.utcnow()
+        # activity logs are only archived for a period of 90 days max (requesting a timespan of more than that ends up
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/facade/storageaccounts.py]:
```suggestion
        # Activity logs are only archived for a period of 90 days max (requesting a timespan of more than that ends up
```

**Pull request merged**